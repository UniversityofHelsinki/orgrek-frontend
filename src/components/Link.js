import React from 'react';
import { Link as RouterLink, useSearchParams } from 'react-router-dom';
import MuiLink from '@mui/material/Link';

/**
 * Use this component to create links within the app.
 *
 * Styled as MUI Link but behaves like a router link.
 * Preserves current query parameters, i.e. current hierarchies are included
 * in the link when linking to another node.
 *
 * @param to router path
 * @param node uid query param
 * @param hierarchies hierarchies query param
 * @param sx The system prop that allows defining system overrides as well as additional CSS styles.
 * @param props any other props passed to MUI and router Link
 */
const Link = ({ to, node, hierarchies, sx, ...props }) => {
  const [searchParams] = useSearchParams();

  const toParams = new URLSearchParams(searchParams.toString());

  if (hierarchies) {
    toParams.set('hierarchies', hierarchies);
  }

  if (node) {
    toParams.set('uid', node);
  }

  return (
    <MuiLink
      component={RouterLink}
      {...props}
      sx={{
        fontWeight: 'medium',
        ':hover': { color: 'primary.dark' },
        ...sx,
      }}
      to={`${to || ''}?${toParams.toString()}`}
    />
  );
};

export default Link;
