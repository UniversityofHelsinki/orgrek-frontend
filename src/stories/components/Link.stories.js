import React from 'react';
import Link from '../../components/Link';

export default {
  component: Link,
  tags: ['autodocs'],
};

export const NodeLink = {
  args: {
    to: '',
    node: 12345678,
    hierarchies: '',
    text: 'Helsingin yliopisto (HY)',
  },
  render: ({ text, ...args }) => <Link {...args}>{text}</Link>,
};

export const HierarchiesLink = {
  ...NodeLink,
  args: {
    to: '',
    node: '',
    hierarchies: 'talous,opetus',
    text: 'Hierarchy link',
  },
};

export const PathLink = {
  ...NodeLink,
  args: {
    to: '/texts',
    node: '',
    hierarchies: '',
    text: 'Texts',
  },
};
