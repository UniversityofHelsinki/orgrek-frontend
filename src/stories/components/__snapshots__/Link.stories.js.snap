// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`stories/components/Link HierarchiesLink smoke-test 1`] = `
Object {
  "checked": undefined,
  "children": Array [
    Object {
      "checked": undefined,
      "children": undefined,
      "name": "Hierarchy link",
      "pressed": undefined,
      "role": "link",
      "value": undefined,
    },
  ],
  "name": "stories-components-link--hierarchies-link",
  "pressed": undefined,
  "role": "WebArea",
  "value": undefined,
}
`;

exports[`stories/components/Link NodeLink smoke-test 1`] = `
Object {
  "checked": undefined,
  "children": Array [
    Object {
      "checked": undefined,
      "children": undefined,
      "name": "Helsingin yliopisto (HY)",
      "pressed": undefined,
      "role": "link",
      "value": undefined,
    },
  ],
  "name": "stories-components-link--node-link",
  "pressed": undefined,
  "role": "WebArea",
  "value": undefined,
}
`;

exports[`stories/components/Link PathLink smoke-test 1`] = `
Object {
  "checked": undefined,
  "children": Array [
    Object {
      "checked": undefined,
      "children": undefined,
      "name": "Texts",
      "pressed": undefined,
      "role": "link",
      "value": undefined,
    },
  ],
  "name": "stories-components-link--path-link",
  "pressed": undefined,
  "role": "WebArea",
  "value": undefined,
}
`;
